revocation/revocation.go

```diff
@@ -33,10 +35,29 @@ type Revocation interface {
 	Validate(certChain []*x509.Certificate, signingTime time.Time) ([]*result.CertRevocationResult, error)
 }
 
+// ContextRevocation is an interface that specifies methods used for revocation
+// checking with context
+type ContextRevocation interface {
+	// ValidateContext checks the revocation status for a certificate chain
+	// and returns an array of CertRevocationResults that contain the results
+	// and any errors that are encountered during the process
+	ValidateContext(ctx context.Context, certChain []*x509.Certificate, signingTime time.Time) ([]*result.CertRevocationResult, error)
```

Comment:
We are creating a new interface, so I want to mention the `revocation mode` feature. Although we will not include it in v1.3.0, will we need it in the future? If we need it, we may consider it now. For `revocation mode`, it cannot be an option for `NewWithOptions` because when we create the revocation client, we don't know which trust policy will be selected, which is based on trust scope. Therefore, the `revocation mode` should be passed in when actually doing the revocation, that is, when calling the `ValidateContext` function. So, should we include `revocation mode` in this function?

Reply:
`revocation mode` is for selecting the rule of revocation, such as automatic fallback, only OCSP, or only CRL.

Reply:
Can we provide extensibility for this function. For example
```suggestion
	ValidateContext(ctx context.Context, certChain []*x509.Certificate, signingTime time.Time, opts *ValidateOptions) ([]*result.CertRevocationResult, error)
```

Reply:
cc: @shizhMSFT

Reply:
@JeyJeyGao I think that's a good idea. We should provide extensibility to the new interface. Let me update the PR.

Reply:
Updated.